PrebidMobile/DemandSDKAdaptersForDFP/src/main/java/org/prebid/demandsdkadapters/dfp/PrebidCustomEventBanner.java

```diff
@@ -0,0 +1,111 @@
+package org.prebid.demandsdkadapters.dfp;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+
+import com.google.android.gms.ads.AdRequest;
+import com.google.android.gms.ads.AdSize;
+import com.google.android.gms.ads.mediation.MediationAdRequest;
+import com.google.android.gms.ads.mediation.customevent.CustomEventBanner;
+import com.google.android.gms.ads.mediation.customevent.CustomEventBannerListener;
+
+import org.prebid.demandsdkadapters.common.AdListener;
+import org.prebid.demandsdkadapters.common.BannerController;
+import org.prebid.mobile.core.LogUtil;
+import org.prebid.mobile.core.PrebidDemandSettings;
+import org.prebid.mobile.core.ErrorCode;
+
+import static org.prebid.mobile.core.PrebidDemandSettings.FACEBOOK_BIDDER_NAME;
+import static org.prebid.mobile.core.PrebidDemandSettings.PREBID_BIDDER;
+import static org.prebid.mobile.core.PrebidDemandSettings.PREBID_CACHE_ID;
+import static org.prebid.mobile.core.PrebidDemandSettings.isDemandEnabled;
+
+public class PrebidCustomEventBanner implements CustomEventBanner, AdListener {
+    private BannerController controller;
+    private CustomEventBannerListener listener;
+
+    @Override
+    public void requestBannerAd(Context context, CustomEventBannerListener customEventBannerListener, String s, AdSize adSize, MediationAdRequest mediationAdRequest, Bundle bundle) {
+        LogUtil.i("Requesting custom event banner through DFP.");
+        this.listener = customEventBannerListener;
+        if (bundle != null) {
+            String cacheId = (String) bundle.get(PREBID_CACHE_ID);
```

Comment:
Do we need to make sure cacheId isn't null here? For example, is there a case where the pub forgets to add the bundle and therefore we aren't able to add the cacheId and the line item still gets here?

Reply:
If pub forgets to add the bundle then bundle would be null this won't be called.